File: kubik/kubik-server/src/main/webapp/public/js/warehouse/stocktaking-service.js
(function(){
	var STOCKTAKING_URL = "/stocktaking";
	angular
		.module("Kubik")
		.factory("stocktakingService", StocktakingService);
	
	function StocktakingService($http){
		return {
			findAll : findAll,
			findOne : findOne,
			generateDiff : generateDiff, 
			save : save,
			updateInventory : updateInventory
		};
		
		function findAll(){
			return $http
				.get(STOCKTAKING_URL)
				.then(findAllSuccess);
			
			function findAllSuccess(response){
				return response.data; 
			}
		}
		
		function findOne(id){
			return $http
				.get(STOCKTAKING_URL + "/" + id)
				.then(findOneSuccess);
			
			function findOneSuccess(response){
				return response.data;
			}
		}
		
		function generateDiff(id){
			return $http
				.post(STOCKTAKING_URL + "/" + id + "?generateDiffs");
		}
		
		function save(stocktaking){
			return $http
				.post(STOCKTAKING_URL, stocktaking)
				.then(saveSuccess);
			
			function saveSuccess(response){
				return response.data;
			}
		}
		
		function updateInventory(stocktakingId){
			return $http
				.post(STOCKTAKING_URL + "/" + stocktakingId + "?updateInventory")
				.then(postSuccess);
			
			function postSuccess(response){
				return response.data;
			}
		}
	}
})();

File: kubik/kubik-server/src/main/java/com/cspinformatique/kubik/server/domain/warehouse/service/StocktakingService.java
package com.cspinformatique.kubik.server.domain.warehouse.service;

import java.util.List;

import com.cspinformatique.kubik.server.model.product.Product;
import com.cspinformatique.kubik.server.model.warehouse.Stocktaking;

public interface StocktakingService {

	void applyInventoryAdjustments(long id);

	void applyInventoryAdjustments(Product product, double addedQuantity);

	List<Stocktaking> findAll();

	Stocktaking findOne(long id);

	Stocktaking generateDummyStocktaking();

	void generateStocktakingDiffs(long id);

	Stocktaking save(Stocktaking stocktaking);

	Stocktaking updateInventory(long stocktakingId);
}
